liquibase-standard/src/main/java/liquibase/sqlgenerator/core/InsertGenerator.java

```diff
@@ -94,7 +96,11 @@ public void appendValue(StringBuilder sql, Database database, Object newValue) {
                 sql.append(DataTypeFactory.getInstance().getFalseBooleanValue(database));
             }
         } else if (newValue instanceof DatabaseFunction) {
-            sql.append(database.generateDatabaseFunctionValue((DatabaseFunction) newValue));
+            if (newValue instanceof SequenceNextValueFunction && database instanceof HsqlDatabase) {
+                sql.append("NEXT VALUE FOR ").append(((SequenceNextValueFunction) newValue).getValue());
+            } else {
+                sql.append(database.generateDatabaseFunctionValue((DatabaseFunction) newValue));
+            }
```

Comment:
Would it be better to make this an abstract/default method on either the database or whatever class `SequenceNextValueFunction` implements/extends?

Reply:
We have `database.sequenceNextValueFunction` that is used by `database.generateDatabaseFunctionValue` in a generic way, but sadly seems HSQL database it's the only db that uses different expressions to do the samething when used in different contexts. So I would go with this if now, but _if_ another db needs it we can create another  abstract/default method to handle it.